OrbitService/Process.cpp

```diff
@@ -19,7 +19,22 @@
 
 namespace orbit_service {
 
-ErrorMessageOr<Process> Process::FromPidAndCpuUsage(pid_t pid, double cpu_usage) {
+void Process::UpdateCpuUsage(utils::Jiffies process_cpu_time, utils::Jiffies total_cpu_time) {
+  const auto diff_process_cpu_time =
+      static_cast<double>(process_cpu_time.value - previous_process_cpu_time_.value);
+  const auto diff_total_cpu_time =
+      static_cast<double>(total_cpu_time.value - previous_total_cpu_time_.value);
+
+  const auto cpu_usage = std::clamp(diff_process_cpu_time / diff_total_cpu_time, 0.0, 1.0);
```

Comment:
can this be less than 0?

Reply:
No, it can't because Jiffies is unsigned. But it can be bigger than 1.0 when the counter wraps.

But if Jiffies were signed we would get a negative values in the wrapping case. So I had two choices: Either add a static_assert which ensure that Jiffies is unsigned or add the clamp. I felt that the clamp is less surprising. It's also part of all the examples I've on the internet. (top has clamp, htop as well, the stuff on StackOverflow also has it.)

Do you think it's misleading?

Reply:
no but I think it deserves a short comment as of when it can be > 1.0

Reply:
Yes, that makes sense. Done. And thank you!